v2/clients/http/utils/request.go

```diff
@@ -127,8 +128,13 @@ func PutByFileRequest(
 }
 
 // Helper method to make the delete request and return the body
-func DeleteRequest(ctx context.Context, returnValuePointer interface{}, url string) errors.EdgeX {
-	req, err := createRequest(ctx, http.MethodDelete, url)
+func DeleteRequest(ctx context.Context, returnValuePointer interface{}, baseUrl string, requestPath string) errors.EdgeX {
+	u, err := url.Parse(baseUrl)
+	if err != nil {
+		return errors.NewCommonEdgeX(errors.KindClientError, "fail to parse baseUrl", err)
+	}
+	u.Path = requestPath
+	req, err := createRequest(ctx, http.MethodDelete, u.String())
```

Comment:
Seems that the common code dealing with parsing and building URL should be in the common createRequest so it isn't repeated. This can be resolved in a follow-up refactoring issue/PR.

Reply:
Rebased and refactor the createRequest method.